apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/InterestRateModelGraph/components/Chart/Chart.tsx

```diff
@@ -1,27 +1,69 @@
-import React, { FC, useCallback, useEffect, useRef } from 'react';
+import React, { FC, useCallback, useEffect, useMemo, useRef } from 'react';
 
 import ChartLibrary from 'chart.js/auto';
 import 'chartjs-adapter-date-fns';
 
 import { theme } from '@sovryn/tailwindcss-config';
 
 import {
+  CHART_PERCENTAGES,
   CUSTOM_CANVAS_BACKGROUND_COLOR,
   GRID_COLOR,
   TICK_COLOR,
 } from './Chart.constants';
-import { MockData } from './Chart.types';
-import { htmlLegendPlugin } from './Chart.utils';
+import { RatesData } from './Chart.types';
+import {
+  calculateVariableInterestRateModel,
+  htmlLegendPlugin,
+} from './Chart.utils';
 
 type ChartProps = {
-  mockData: MockData<{ x: number; y: number }>;
-  yLabel1: string;
+  meta: {
+    label: string;
+    lineColor: string;
+  };
+  rates: RatesData;
 };
 
-export const Chart: FC<ChartProps> = ({ mockData }) => {
+export const Chart: FC<ChartProps> = ({ meta, rates }) => {
   const canvas = useRef<HTMLCanvasElement>(null);
   const chartRef = useRef<ChartLibrary | null>(null);
 
+  const variableValues = useMemo(
+    () =>
+      CHART_PERCENTAGES.map(x => ({
+        x: x * 100,
+        y: calculateVariableInterestRateModel(x, rates) * 100,
+      })),
+    [rates],
+  );
+
+  const optimalPercentage = useMemo(() => {
+    return rates.optimalUsageRatio
+      ? parseFloat((parseFloat(rates.optimalUsageRatio) * 100).toFixed(2))
+      : 0;
+  }, [rates.optimalUsageRatio]);
+
+  const currentPercentage = useMemo(() => {
+    return rates.currentUsageRatio
+      ? parseFloat((parseFloat(rates.currentUsageRatio) * 100).toFixed(2))
+      : 0;
+  }, [rates.currentUsageRatio]);
+
+  const optimalValue = useMemo(() => {
+    return [
+      { x: optimalPercentage, y: 0 },
+      { x: optimalPercentage, y: 75 },
+    ];
+  }, [optimalPercentage]);
```

Comment:
is it okay the y hardcoded?

Reply:
I think for now all graphs are mocked so should be fine